### DISCUSSION TITLE
class_labels in UNet3DConditionModel

### DISCUSSION THREAD

**Discussion by KAI10** [huggingface/diffusers#9019] [Q&A] [answered]
Hi, 

I am going through the source code of [UNet3DConditionModel](https://github.com/huggingface/diffusers/blob/main/src/diffusers/models/unets/unet_3d_condition.py), specifically the [forward](https://github.com/huggingface/diffusers/blob/e5b94b4c57e5499a260f6c4a4979e0b55357392e/src/diffusers/models/unets/unet_3d_condition.py#L552) method. Here a parameter named `class_labels` is available with the description: 

_class_labels (`torch.Tensor`, *optional*, defaults to `None`):
Optional class labels for conditioning. Their embeddings will be summed with the timestep embeddings._

However, it seems this parameter is not used in the method. Is this part not implemented yet?

Thanks in advance!

**Comment by asomoza** [comment]
cc @sayakpaul

**Comment by sayakpaul** [comment, marked as answer]
Yes, `class_labels` isn't implemented yet for the 3D variant.
